Telerik.JustMock/Core/MocksRepository.cs

```diff
@@ -415,15 +416,18 @@ internal void DispatchInvocation(Invocation invocation)
 
 			if (!methodMockProcessed)
 			{
+				// We have to be cearfull for the potential exception throwing in the assertion context, so skip CallOriginalBehavior processing
 				var mock = invocation.MockMixin;
 				if (mock != null)
 				{
-					foreach (var behavior in mock.FallbackBehaviors)
-						behavior.Process(invocation);
+                    mock.FallbackBehaviors
+                        .Where(behavior => !invocation.InAssertSet || !(behavior is CallOriginalBehavior))
```

Comment:
It seems that there is no unit test for this scenario when the CallOriginalBehavior should not be processed.

Reply:
@mihail-vladov, it seems that we have such, please see [ShouldNotThrowDuringAssertForCallOriginalWhenNoArrangeSpecified](https://github.com/telerik/JustMockLite/blob/463187c12bdf1b743fad63a4c8ac9188a638a36a/Telerik.JustMock.Tests/AssertionFixture.cs#L407)